src/scheduler.rs

```diff
@@ -715,6 +795,56 @@ impl PodIdentity {
         }
     }
 
+    /// Returns a string with all pod labels required by the [`PodIdentity`] joined with comma.
+    pub fn labels(pod: &Pod) -> SchedulerResult<String> {
+        if pod.metadata.labels.is_none() {
+            return Err(Error::PodWithoutLabelsNotSupported);
+        }
+
+        let mut result: Vec<String> = vec![];
+
+        let pod_labels = &pod.metadata.labels.as_ref().unwrap();
+        for label_name in [
+            labels::APP_NAME_LABEL,
+            labels::APP_INSTANCE_LABEL,
+            labels::APP_COMPONENT_LABEL,
+            labels::APP_ROLE_GROUP_LABEL,
+        ] {
+            result.push(
+                pod_labels
+                    .get(label_name)
+                    .cloned()
+                    .ok_or(Error::PodWithoutLabelsNotSupported)?,
```

Comment:
Would be nice to have a message here about what label is missing.

Reply:
Done.